mesonbuild/coredata.py

```diff
@@ -718,6 +718,13 @@ def set_options(self, options, *, subproject='', warn_unknown=True):
             self.copy_build_options_from_regular_ones()
 
     def set_default_options(self, default_options, subproject, env):
+        # Warn if the user is using two different ways of setting build-type
+        # options that override each other
+        if 'buildtype' in env.cmd_line_options and \
+           ('optimization' in env.cmd_line_options or 'debug' in env.cmd_line_options):
+            mlog.warning('Recommend using either -Dbuildtype or -Doptimization + -Ddebug. '
```

Comment:
I'm not native English speaker, but I don't think "Recommend using" is the proper formulation here. Shouldn't it be "It is recommended to use either ..." ?

Reply:
I'm not a native English speaker either, but as I understand it, here `Recommend using` is a shortened assertive form, similar to what's used in newspaper headlines. It's short for `We recommend using` where `we` is obvious from context. Words like `It is` and `We` are just noise in short messages.

Reply:
ok, if you say so. It doesn't matter much :-)